From 60fd2a0fff8b7b8c041d5ab4856abe2779417c0e Mon Sep 17 00:00:00 2001
From: Esben Zeuthen <esbenzeuthen@gmail.com>
Date: Wed, 29 Jan 2025 22:11:43 +0000
Subject: [PATCH] Hacked for Telink Mesh StartNotify support

---
 src/shared/gatt-client.c | 18 +++++++++++++++++-
 1 file changed, 17 insertions(+), 1 deletion(-)

diff --git a/src/shared/gatt-client.c b/src/shared/gatt-client.c
index 9db3f5211..2031dd269 100644
--- a/src/shared/gatt-client.c
+++ b/src/shared/gatt-client.c
@@ -1685,12 +1685,28 @@ static bool notify_data_write_ccc(struct notify_data *notify_data, bool enable,
 			return false;
 	}
 
-	att_id = bt_gatt_client_write_value(notify_data->client,
+	static const uint8_t telink_mesh_notify_uuid[16] = {0x0, 0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8, 0x9, 0xa, 0xb, 0xc, 0xd, 0x19, 0x11};
+
+	const bt_uuid_t *uuid = gatt_db_attribute_get_type(notify_data->chrc->attr);
+	if (memcmp((const void*)&uuid->value.u128, (const void*)telink_mesh_notify_uuid, sizeof(telink_mesh_notify_uuid)) == 0) {
+		att_id = bt_gatt_client_write_value(notify_data->client,
+						notify_data->chrc->value_handle, //EZ NOTE: changed from ccc_handle as a hack!
+						(void *)&value, sizeof(value),
+						callback,
+						notify_data_ref(notify_data),
+						notify_data_unref);
+	} else
+	{
+			att_id = bt_gatt_client_write_value(notify_data->client,
 						notify_data->chrc->ccc_handle,
 						(void *)&value, sizeof(value),
 						callback,
 						notify_data_ref(notify_data),
 						notify_data_unref);
+	}
+
+
+	
 	notify_data->chrc->ccc_write_id = notify_data->att_id = att_id;
 
 	return !!att_id;
-- 
2.34.1

